crates/view/src/storage.rs

```diff
@@ -1036,6 +1033,12 @@ impl Storage {
             // do it this way; if it becomes slow, we can do it better
             let mut results = Vec::new();
             for record in spendable_note_records {
+                // Skip notes that don't match the account index, if declared.
+                if let Some(address_index) = address_index {
+                    if record.address_index.account != address_index.account {
+                        continue;
+                    }
+                }
```

Comment:
```suggestion
                if matches!(address_index, Some(a) if a.account != record.address_index.account) {
                      continue;
                }
```

Potential improvement?